### PR TITLE
[nis] feat: jenkins docker build, publish integration

### PR DESCRIPTION
**Pull request by yilmazbahadir**
An alternative(probably better) approach to https://github.com/NemProject/nem/pull/191.

What was the issue?
- in https://github.com/NemProject/nem/pull/191 dockerfile was pulling from remote repo, we weren't successful to pass the current branch to the dockerfile.
- it was adding a new jenkins job to the multi-branch jenkins job which was running in any case therefore causing a performance hit.
- In our discussions with @Wayonb we agreed that this can cause issues, better to build from workspace as it's been done for `symbol/rest`.

What's the fix?
- Created an alternative Dockerfile in the `nis` module, which is copying the code from the jenkins workspace _using the current branch_ which will allow us to push images from feature branches for testing. (the same way as `symbol/rest` was configured`)
- Docker publish parameters are placed inside `nis/Jenkinsfile` thus we'll utilize the current job and we won't need to create any other Job like we did in https://github.com/NemProject/nem/pull/191, which puts an overhead to the Jenkins job even we don't publish any docker images.

### CODE REVIEW

File: nis/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+WORKDIR /app
+
+COPY --from=builder /build/nem/infra/package /app
+
+RUN chown -R ${UNAME} /app
+
+USER $UNAME
+
+CMD ["bash", "-c", "java -Xms${MEMORY_MS} -Xmx${MEMORY_MX} -cp /usersettings:./nis/*:./libs/* org.nem.deploy.CommonStarter"]
```
Comment (Wayonb) [inline nis/Dockerfile:47]: missing new line at the end of file

File: nis/Jenkinsfile [hunk 2]
```diff
@@ -2,7 +2,11 @@ defaultCiPipeline {
 	platform = ['ubuntu']
 	ciBuildDockerfile = 'java.Dockerfile'
 
-	packageId = 'nem-nis'
+    publisher = 'docker'
```
Comment (Wayonb) [inline nis/Jenkinsfile:5]: spacing

File: nis/Jenkinsfile [hunk 3]
```diff
@@ -2,7 +2,11 @@ defaultCiPipeline {
...
+	dockerImageName = 'nemofficial/nis-client'
+    dockerBuildArgs = "-f ./Dockerfile  ./../"
```
Comment (Wayonb) [inline nis/Jenkinsfile:7]: maybe this? dockerBuildArgs = "-f Dockerfile .."

**Review by Wayonb — CHANGES_REQUESTED** [review]

**Review by gimre-xymcity — APPROVED** [review]
approving, assuming comments from wayon

**Comment by yilmazbahadir** [comment]
@Wayonb comments should be fixed and also updated the nis/README.md for the docker instructions.

**Review by Wayonb — APPROVED** [review]

**Pull request merged**